File: work/DSA/c++/recursion_1b/towerOfHanoi.cpp
#include <bits/stdc++.h>
using namespace std;

void towerOfHanoi(int n, char source, char auxilary, char destination)
{
    if (n == 0)
    {
        return;
    }
    towerOfHanoi(n - 1, source, destination, auxilary);
    cout << source << " " << destination << endl;
    towerOfHanoi(n - 1, auxilary, source, destination);
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;
    towerOfHanoi(n, 'a', 'b', 'c');
}

File: work/DSA/javascript/recursion_1b/towerOfHanoi.js
let n = 2
function towerOfHanoi(n, source, auxilary, destination) {
    if (n == 0) {
        return
    }
    towerOfHanoi(n - 1, source, destination, auxilary)
    console.log(source + " " + destination)
    towerOfHanoi(n - 1, auxilary, source, destination)
}

towerOfHanoi(n, 'a', 'b', 'c')
